src/wallet/multisig_info.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018-2020, The Loki Project
+// Copyright (c) 2014-2019, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+#include <ringct/rctTypes.h>
+#include <boost/serialization/version.hpp>
+
+namespace wallet {
```

Comment:
Spreading things into individual files where 1 file = 1 thing is entirely the point.  What is the advantage in blobbing them together *in the source code*?  There's no advantage at all to the compiler, and from the human perspective it seems cleaner to me to have 10 files that each does one logical thing (or one set of closely related things) rather than one file that does 10 not-particularly-related things.  Obviously there are limitations to that, but none of these are minuscule, trivial headers, and other than the fact that they are all used in the wallet, each one is different.

> some of them end up re-including each other anyway.

Doesn't seem relevant.  Whether two things belong in the same header depends on what they *do* (i.e. separation of concerns), not how they get used.  For example: `tx_sets.h`'s concern is being a container.  It includes `pending_tx.h` because some of the containers contain pending_tx, but it has nothing to do with the actual inner structure of pending_tx -- because that's `pending_tx.h`'s concern.  Even though `pending_tx.h` is not included anywhere that doesn't also include `tx_sets.h`, that's not a justification for combining those concerns (and similarly for everything else that gets included here.).  I'm fine with "A and B are tightly coupled therefore logically belong in the same header", I'm not fine with "currently A and B always get included at the same time therefore belong in the same header."

Reply:
Organizing either way are both things I'm fine with it does boil down to personal preference for readability, I see the two ways of organizing as the same thing from different views. To use the file system to group or grouping it textually, the thing that leads me to prefer in the file itself is that we use text editors which optimise for one more than the other, of course all in consideration of the little code we're putting across files. There are things that are clearer when the boundary between them is files not text.

It's true the reincluding part was not as circular as I make it out to be.

I do note we end up in one of the situations where you only need to include one of the header files instead of the others and that's nice yes and has benefits, no down-playing that.